lib/src/interface/local_process_manager.dart

```diff
@@ -99,7 +98,79 @@ class LocalProcessManager implements ProcessManager {
   }
 
   @override
+  bool canRun(@checked Object executable, {String workingDirectory}) =>
+      getExecutablePath(executable, workingDirectory) != null;
+
+  @override
   bool killPid(int pid, [ProcessSignal signal = ProcessSignal.SIGTERM]) {
     return Process.killPid(pid, signal);
   }
 }
+
+String _getExecutable(
+    List<dynamic> command, String workingDirectory, bool runInShell) {
+  String commandName = command.first.toString();
+  if (runInShell) {
+    return commandName;
+  }
+  String exe = getExecutablePath(commandName, workingDirectory);
+  if (exe == null) {
+    throw new ArgumentError('Cannot find executable for $commandName.');
+  }
+  return exe;
+}
+
+List<String> _getArguments(List<dynamic> command) =>
+    command.skip(1).map((dynamic element) => element.toString()).toList();
+
+/// Searches the `PATH` for the actual executable that [commandName] is supposed
+/// to launch.
+///
+/// Return `null` if the executable cannot be found.
+@visibleForTesting
+String getExecutablePath(String commandName, String workingDirectory,
```

Comment:
Put in a separate `common.dart` that's not exported by the library -- so the test can still import it under lib/src, but package users don't get it by accident.